mp/src/game/server/momentum/mom_player.cpp

```diff
@@ -1096,12 +1096,18 @@ void CMomentumPlayer::OnZoneEnter(CTriggerZone *pTrigger)
     case ZONE_TYPE_STOP:
         {
             // We've reached end zone, stop here
-            //auto pStopTrigger = static_cast<CTriggerTimerStop *>(pTrigger);
+            const auto pStopTrigger = static_cast<CTriggerTimerStop *>(pTrigger);
             m_iOldTrack = m_Data.m_iCurrentTrack;
             m_iOldZone = m_Data.m_iCurrentZone;
 
             if (g_pMomentumTimer->IsRunning())
             {
+                if(pStopTrigger->GetCancelBool())
+                {
+                    g_pMomentumTimer->Reset(this);
```

Comment:
You should instead use:

```suggestion
                    g_pMomentumTimer->Stop(this);
```

As this will stop the timer and make the timer flash red to indicate it was stopped. Reset is only done in the Start zone, as it sets `m_bCanStart` to true, which would be pretty bad here.

Reply:
Also, spacing between the if and the parenthesis above this.